madprops/serializers.py

```diff
@@ -52,6 +53,12 @@ def to_representation(self, value):
         return result_dict
 
     def to_internal_value(self, data):
+        try:
+            data = dict(data)
+        except ValueError:
+            data_type = type(data).__name__
+            raise ValidationError(
```

Comment:
Given that this bit of code has no concept of how the data was provided (ie. JSON, YAML, XML, Carrier Pigeon, etc), does it make sense to explicitly utilize JSON/JS terminology? Should something more generic like `mapping`, or `key value hash`, or ?? be provided?